File: Advanced DSA/02. Array Techniques/03. Sum of All Subarrays/Solution.cpp
long Solution::subarraySum(vector<int> &A) {
    int n = A.size();
    long total_sum = 0;
    for (int i = 1; i <= n; ++i)
        total_sum += (A[i - 1] * i * (n - i + 1L));
    return total_sum;
}

File: Advanced DSA/02. Array Techniques/03. Sum of All Subarrays/Solution.py
class Solution:
    # @param A : list of integers
     # @return an long
    def subarraySum(self, A):
        n = len(A)
        total_sum = 0
        for i in range(1, n + 1):
            total_sum += (A[i - 1] * i * (n - i + 1))
        return total_sum
